src/counts.jl

```diff
@@ -238,12 +238,48 @@ If a weighting vector `wv` is specified, the sum of the weights is used rather t
 raw counts.
 """
 function addcounts!(cm::Dict{T}, x::AbstractArray{T}) where T
+    # if it's of bits type then can speed things up using radix sort
+    # from heuristics it was found that 2^16 = 65536 is the point at which
+    # radixsort is more performant
+    if isbits(T) & length(x) > 65536
+        addcounts_radixsort!(cm, x)
+        return cm
+    else
+        addcounts_dict!(cm,x)
+        return cm
+    end
+end
+
+function addcounts_dict!(cm::Dict{T}, x::AbstractArray{T}) where T
     for v in x
         cm[v] = get(cm, v, 0) + 1
```

Comment:
This algorithm isn't optimal, it would be possible to use `ht_keyindex` to avoid a double lookup (as I noted on Discourse). Maybe the 65536 threshold would be different if you do that?

Reply:
I thought about doing but I thought the API for Dict might change to include updateindex! to update the value once found in one lookup. Is ht_keyindex going to be a long term solution?

Reply:
Unfortunately not, but `updateindex!` won't be added in 1.0 AFAIK. So I'd rather at least know what's the best performance we can achieve with `Dict` before adding a new radix-based algorithm.